stack/services/src/main/java/org/apache/usergrid/services/notifications/wns/WNSAdapter.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.usergrid.persistence.EntityManager;
+import org.apache.usergrid.persistence.entities.Notification;
+import org.apache.usergrid.persistence.entities.Notifier;
+import org.apache.usergrid.services.ServicePayload;
+import org.apache.usergrid.services.notifications.ConnectionException;
+import org.apache.usergrid.services.notifications.ProviderAdapter;
+import org.apache.usergrid.services.notifications.TaskTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Windows Notifications Service adapter to send windows notifications
+ */
+public class WNSAdapter implements ProviderAdapter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WNSAdapter.class);
+
+    private final EntityManager entityManager;
+    private final Notifier notifier;
+    private final WnsService service;
+
+    public WNSAdapter(EntityManager entityManager, Notifier notifier) {
+        this.entityManager = entityManager;
+        this.notifier = notifier;
+        this.service = new WnsService(notifier.getSid(), notifier.getApiKey(), notifier.getLogging());
+    }
+
+    @Override
+    public void testConnection() throws ConnectionException {
+        WnsToast toast = new WnsToastBuilder().bindingTemplateToastText01("test").build();
+        try{
+            //this fails every time due to jax error which is ok
+            service.pushToast("s-1-15-2-2411381248-444863693-3819932088-4077691928-1194867744-112853457-373132695",toast);
+        }catch (ClientHandlerException e){
+            LOG.info("Windows Phone notifier added: "+e.toString());
+        }
+    }
+
+    @Override
+    public void sendNotification(String providerId, Object payload, Notification notification, TaskTracker tracker) throws Exception {
+        try {
+            List<TranslatedNotification> translatedNotifications = ( List<TranslatedNotification>) payload;
+            for(TranslatedNotification translatedNotification : translatedNotifications) {
+                switch (translatedNotification.getType()) {
+                    case "toast":
+                        WnsToast toast = new WnsToastBuilder().bindingTemplateToastText01(translatedNotification.getMessage().toString()).build();
+                        service.pushToast(providerId, toast);
+                        break;
+                    case "badge":
+                        WnsBadge badge;
+                        if (translatedNotification.getMessage() instanceof Integer) {
+                            badge = new WnsBadgeBuilder().value((Integer) translatedNotification.getMessage()).build();
+                        } else {
+                            badge = new WnsBadgeBuilder().value(translatedNotification.getMessage().toString()).build();
+                        }
+                        service.pushBadge(providerId, badge);
+                        break;
```

Comment:
Same thing here, do we want to throw/log an exception for when it's not a type we handle?  That would be helpful operationally if users are having problems and can't send.